storagemarket/impl/provider_environments.go

```diff
@@ -45,24 +76,50 @@ func (p *providerDealEnvironment) Ask() storagemarket.StorageAsk {
 	return *sask.Ask
 }
 
-func (p *providerDealEnvironment) DeleteStore(storeID multistore.StoreID) error {
-	return p.p.multiStore.Delete(storeID)
+func (p *providerDealEnvironment) CleanReadWriteBlockstore(proposalCid cid.Cid, carV2FilePath string) error {
+	// close the backing CARv2 file and stop tracking the read-write blockstore for the deal with the given proposalCid.
+	if err := p.p.readWriteBlockStores.CleanBlockstore(proposalCid.String()); err != nil {
+		log.Warnf("failed to clean read write blockstore, proposalCid=%s, carV2FilePath=%s, err=%s", proposalCid, carV2FilePath, err)
+	}
+
+	// clean up the backing CARv2 file.
+	return p.p.fs.Delete(filestore.Path(carV2FilePath))
 }
 
-func (p *providerDealEnvironment) GeneratePieceCommitment(storeID *multistore.StoreID, payloadCid cid.Cid, selector ipld.Node) (cid.Cid, filestore.Path, error) {
-	proofType, err := p.p.spn.GetProofType(context.TODO(), p.p.actor, nil)
+// GeneratePieceCommitment generates the pieceCid for the CARv1 deal payload in the CARv2 file that already exists at the given path.
+func (p *providerDealEnvironment) GeneratePieceCommitment(proposalCid cid.Cid, carV2FilePath string) (cid.Cid, filestore.Path, error) {
+	rd, err := carv2.NewReaderMmap(carV2FilePath)
 	if err != nil {
-		return cid.Undef, "", err
+		return cid.Undef, "", xerrors.Errorf("failed to get CARv2 reader, proposalCid=%s, carV2FilePath=%s, err=%w", proposalCid, carV2FilePath, err)
 	}
-	if p.p.universalRetrievalEnabled {
-		return providerutils.GeneratePieceCommitmentWithMetadata(p.p.fs, p.p.pio.GeneratePieceCommitment, proofType, payloadCid, selector, storeID)
+
+	defer func() {
+		if err := rd.Close(); err != nil {
+			log.Errorf("failed to close CARv2 reader, proposalCid%s, carV2FilePath=%s, err=%s", proposalCid, carV2FilePath, err)
```

Comment:
Why not return the error from this function?

Reply:
This is fixed.